pkg/k8shandler/collection.go

```diff
@@ -219,6 +219,13 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateCollectorServiceAccou
 		delfinalizer = true
 	}
 
+	// If needed create a custom SecurityContextConstraints object for the log collector to run with
+	scc := NewSCC(LogCollectorSCCName)
```

Comment:
@ewolinetz @periklis  we can create this resource statically by declaring it in the manifest can we not?  Is there any reason we should reconcile?  I think there are probably other similar resources where we would prefer that pattern as well.  cc @alanconway

Reply:
I don't think that is something that the OLM can provide currently
cc @ecordell to confirm

Out of curiosity what are we trying to get out of using a new SCC that we can't currently achieve using the currently provided SCCs and roles/bindings on our fluentd SA?

Reply:
@ewolinetz The combination of settings: https://github.com/openshift/cluster-logging-operator/blob/ecaf4daed4e99b22e46d172f86db7980fcf4bb13/pkg/k8shandler/scc.go#L23:L46

Reply:
Trying to further restrict the collector to a minimal set of permissions

Reply:
right i read through that, can we not achieve this using roles and bindings and the currently existing SCCs though?

Reply:
@ewolinetz SCCs provide a backstop for the security settings on the Pod/container level. And they are the only way to control some of these settings.
The existing suitable SCCs are too permissive if we are to really apply the least privilege principle. For example, Fluentd needs to access host volumes, so out of the built-in SCCs we can use privileged, hostaccess or hostmount-anyuid. All of these allow more volume types than fluentd needs access to, for example, persistentVolumeClaim.

scc/privileged and scc/hostaccess don't isolate the IPC, PID and network namespaces for the containers that use them:
```
allowHostIPC: true
allowHostNetwork: true
allowHostPID: true
allowHostPorts: true
```
We can't control these at the Pod or container level with PodSecurityContext or SecurityContext, so that leaves us only scc/hostmount-anyuid as a viable candidate.

Reply:
Tried plugging in scc/hostmount-anyuid as the SCC for fluentd, it didn't work. What's biting us here is this part:
```
apiVersion: security.openshift.io/v1
kind: SecurityContextConstraints
metadata:
 name: hostmount-anyuid
...
seLinuxContext:
  type: MustRunAs
...
```
This requires that fluentd runs with the SELinux context assigned to the project it's running in. In my case, this happens to be
```
apiVersion: project.openshift.io/v1
kind: Project
metadata:
...
    openshift.io/sa.scc.mcs: s0:c26,c5
...
```
and trying to run fluentd fails with the following error

```
provider hostmount-anyuid: .spec.securityContext.seLinuxOptions.type: Invalid value: "spc_t": must be  spec.containers[0].securityContext.seLinuxOptions.level: Invalid value: "": must be s0:c26,c5
```
, plus less obvious errors about not being able to mount hostPath volumes. We have to run fluentd with the `spc_t` SELinux type (super privileged container) because we need to access container logs under `/var/log/containers` and other host mounts. Without `spc_t`, fluentd starts, but is useless:

```
$ oc rsh pod/fluentd-fpthz
# fluentd and the interactive shell are running with the same credentials, note sa.scc.mcs from the openshift-logging project
sh-4.4# ps uZ $$ 1
LABEL                           USER         PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
system_u:system_r:container_t:s0:c5,c26 root   1  0.1  0.2 151012 59460 ?        Ssl  03:21   0:00 /usr/bin/ruby /usr/local/bin/fluentd --suppre
system_u:system_r:container_t:s0:c5,c26 root  35  0.0  0.0  12024  3272 pts/0    Ss   03:22   0:00 /bin/sh
sh-4.4# ls -l /var/log/containers
ls: cannot open directory '/var/log/containers': Permission denied
sh-4.4# ls -ldZ /var/log/containers
drwxr-xr-x. 2 root root system_u:object_r:var_log_t:s0 24576 Jun  7 03:23 /var/log/containers
sh-4.4# 

```

Reply:
> and trying to run fluentd fails with the following error

the error told you what was wrong with that... did you adjust your fluentd pod spec at all?
`spec.containers[0].securityContext.seLinuxOptions.level: Invalid value: "": must be s0:c26,c5`

Reply:
I in fact did, that's what I'm talking about just above starting from "Without spc_t, fluentd starts, but is useless:"